### PR TITLE
modularize test fixtures into their own files

### PR DESCRIPTION
**Pull request by twentylemon**
##### Summary 

`conftest.py` is getting pretty bloated, and it'll only get bigger as we revisit slash commands or make heavier use of discord features, or other librares. This refactors the fixtures a bit, moving them to a new python module `tests.fixtures` and `tests.fixtures.discord`.

There's a few minor changes as well, which I'll try to enumerate:
* remove _if general channel exists_ check in day of the week announcement
* `Insights.check_should_respond` renamed to `check_should_respond_loop`, renamed `__check_should_respond` to just `check_should_respond`; just for easier to read tests
    * similarly, renamed the `__insight` method there to simply `insight`
* removed `setup_general_channel` fixtures (used by day announcement and insights tests); replaced by a `general_channel` fixture instead, which performs the same setup, but also returns the general channel
    * used `general_channel` fixture in those tests as well
*  added `skip_if_private_channel` fixture to skip tests when the `channel` fixture is a group channel or DM
    * really only used for `!start` and `!stop` now, but figured it may be useful for stuff in the future
    * removed `guild_message` and `guild_context`, `skip_if_private_channel` is a more explicit way of doing the same thing

##### Checklist

* [x] this is a source code change
  * [x] run `isort . && black .` in the repository root
  * [x] run `pytest` in the repository root
  * [x] link to issue being fixed using a [_fixes keyword_](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue), if such an issue exists
  * [x] update the wiki documentation if necessary
* [ ] or, this is **not** a source code change

### CODE REVIEW

**Comment by twentylemon** [comment]
> Merging [#311](https://codecov.io/gh/Chippers255/duckbot/pull/311?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Thomas+Nelson) ([8c78270](https://github.com/Chippers255/duckbot/commit/8c78270b2674dfcb543a17f77946c0539fc82b50)) into [main](https://codecov.io/gh/Chippers255/duckbot/commit/da5cf60f5be667d08d695f0caf969a115acee54b?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Thomas+Nelson) ([da5cf60](https://github.com/Chippers255/duckbot/commit/da5cf60f5be667d08d695f0caf969a115acee54b)) will **decrease** coverage by `0.09%`.
> The diff coverage is `75.00%`.

75 is lower than the target 85, the drop is coming from [here](https://app.codecov.io/gh/Chippers255/duckbot/compare/311/changes#D1L51), an `if channel:` check when searching for the general channel. The negative case is no longer covered since I changed the test to only use the `general_channel` fixture instead of all channel types.

Let me know if you'd rather it be totally covered.

**Comment by twentylemon** [comment]
> > Merging [#311](https://codecov.io/gh/Chippers255/duckbot/pull/311?src=pr&el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Thomas+Nelson) ([8c78270](https://github.com/Chippers255/duckbot/commit/8c78270b2674dfcb543a17f77946c0539fc82b50)) into [main](https://codecov.io/gh/Chippers255/duckbot/commit/da5cf60f5be667d08d695f0caf969a115acee54b?el=desc&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=Thomas+Nelson) ([da5cf60](https://github.com/Chippers255/duckbot/commit/da5cf60f5be667d08d695f0caf969a115acee54b)) will **decrease** coverage by `0.09%`.
> > The diff coverage is `75.00%`.
> 
> 75 is lower than the target 85, the drop is coming from [here](https://app.codecov.io/gh/Chippers255/duckbot/compare/311/changes#D1L51), an `if channel:` check when searching for the general channel. The negative case is no longer covered since I changed the test to only use the `general_channel` fixture instead of all channel types.
> 
> Let me know if you'd rather it be totally covered.

Eh. Ended up just removing the `if`, matching the logic we do in the `insights` cog when getting the `#general` channel.

File: tests/conftest.py [hunk 1]
```diff
@@ -1,206 +1,12 @@
...
-    def embed_equals(self, other):
-        return self.to_dict() == other.to_dict()
-
-    def embed_str(self):
-        return str(self.to_dict())
-
-    discord.Embed.__eq__ = embed_equals
-    discord.Embed.__str__ = embed_str
-    discord.Embed.__repr__ = embed_str
-    yield
```
Comment (jameshughes89) [inline tests/conftest.py:206]: wowza

**Review by jameshughes89 — APPROVED** [review]
wowza

**Pull request merged**